lib/Demangling/NodePrinter.cpp

```diff
@@ -1704,10 +1704,16 @@ NodePointer NodePrinter::print(NodePointer Node, bool asPrefixContext) {
     Printer << ": ";
     print(Node->getChild(2));
     return nullptr;
-  case Node::Kind::AssociatedTypeDescriptor:
+  case Node::Kind::AssociatedTypeDescriptor: {
     Printer << "associated type descriptor for ";
-    print(Node->getChild(0));
+    auto assoc = Node->getChild(0);
+    if (auto parent = getFirstChildOfKind(assoc, Node::Kind::Type)) {
```

Comment:
It’s kind of strange that you need to reach into a grandchild node here. Would it make sense to modify the way the child node is printed instead?

Reply:
Sure, that's actually better. I have made the change in feef2da.